### DISCUSSION TITLE
[Enhancement] Better Desktop Experience

### DISCUSSION THREAD

**Discussion by mhrastegari** [dotnet/maui#98] [General]
## Summary
Since WPF,UWP and macOS (sadly Linux still maintained by community) are officially maintained by Microsoft plus there are iPad and Android tablets with keyboard now,
we should have a better things for their user experience.

## API Changes
Like these from UWP :

- [Keyboard Events](https://docs.microsoft.com/en-us/windows/uwp/design/input/keyboard-events)
- [TabIndex](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.control.tabindex?view=winrt-18362)
- [ToolTips](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tooltips)
- [Adaptive MasterPage](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/navigationview#adaptive-behavior)

And these :
- [Context Menu](https://github.com/xamarin/Xamarin.Forms/issues/5883)
- [LongPress/RightClick Events](https://github.com/xamarin/Xamarin.Forms/issues/3480) 

some docs of Android :
- [Handle keyboard actions](https://developer.android.com/training/keyboard-input/commands)
- [keyboard navigation](https://developer.android.com/training/keyboard-input/navigation.html#Tab)
- [Tooltips](https://developer.android.com/guide/topics/ui/tooltips)
- nothing found for Adaptive MasterPage :(

## Platforms
Core/Xaml(All Platforms)
Windows (UWP/WPF)
macOS
Android
iOS

**Comment by ziaulhasanhamim** [comment]
Context menu a deal-breaker. Almost every desktop application uses a context menu for extra options. I like the new desktop top menubar. After top menubar, the context menu should be the first one to support

**Comment by mouralabank** [comment]
Any update about the keyboards?
